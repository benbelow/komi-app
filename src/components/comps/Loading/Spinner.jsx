// @flow
import React from 'react'
import styled, { keyframes } from 'styled-components'

const cubemove = keyframes`
25% { 
  transform: translateX(60px) rotate(-90deg) scale(0.4);
} 50% { 
  transform: translateX(60px) translateY(60px) rotate(-179deg);
} 50.1% { 
  transform: translateX(60px) translateY(60px) rotate(-180deg);
} 75% { 
  transform: translateX(0px) translateY(60px) rotate(-270deg) scale(0.4);
} 100% { 
  transform: rotate(-360deg);
}
`

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`

const Spinner = styled.div`
  margin: 100px auto;
  width: 80px;
  height: 80px;
  position: relative;
`

const Cube = styled.div`
  background-color: #ba68c8;
  border-radius: 25%;
  width: 14px;
  height: 16px;
  position: absolute;
  bottom: 0;
  top: 0;

  animation: ${cubemove} 2s infinite ease-in-out;
`

const Cube1 = styled(Cube)``
const Cube2 = styled(Cube)`
  animation-delay: -0.5s;
  width: 16px;
  height: 14px;
`
const Cube3 = styled(Cube)`animation-delay: -1s;`
const Cube4 = styled(Cube)`
  animation-delay: -1.5s;
  width: 16px;
  height: 14px;
`

export default () => (
  <Wrapper>
    <Spinner>
      <Cube1 />
      <Cube2 />
      <Cube3 />
      <Cube4 />
    </Spinner>
  </Wrapper>
)
