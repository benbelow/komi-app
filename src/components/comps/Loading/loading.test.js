import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import Spinner from './Spinner'

describe('loading test suite', () => {
  test('"Spinner" should match snapshot', () => {
    const comp = renderer.create(<Spinner />)
    const tree = comp.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
