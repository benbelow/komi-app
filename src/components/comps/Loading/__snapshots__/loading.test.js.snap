// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading test suite "Spinner" should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  margin: 100px auto;
  width: 80px;
  height: 80px;
  position: relative;
}

.c2 {
  background-color: #9575CD;
  border-radius: 25%;
  width: 14px;
  height: 16px;
  position: absolute;
  bottom: 0;
  top: 0;
  -webkit-animation: dWlLJE 2s infinite ease-in-out;
  animation: dWlLJE 2s infinite ease-in-out;
}

.c3 {
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
  width: 16px;
  height: 14px;
}

.c4 {
  -webkit-animation-delay: -1s;
  animation-delay: -1s;
}

.c5 {
  -webkit-animation-delay: -1.5s;
  animation-delay: -1.5s;
  width: 16px;
  height: 14px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c3 c2"
    />
    <div
      className="c4 c2"
    />
    <div
      className="c5 c2"
    />
  </div>
</div>
`;
